sdk/Sdk.Generators/Extensions/StringExtensions.cs

```diff
@@ -27,27 +27,26 @@ public static string TrimStringsFromEnd(this string str, IReadOnlyList<string> s
                 result = result.TrimStringFromEnd(s);
             }
 
-            return result; 
+            return result;
         }
 
-        public static string UppercaseFirst(this string str)
+        public static string LowercaseFirst(this string str)
         {
             // Check for empty string.
             if (string.IsNullOrEmpty(str))
             {
                 return string.Empty;
             }
 
-            if (!char.IsUpper(str[0]))
+            if (!char.IsLower(str[0]))
             {
                 // Return char and concat substring.
-                return char.ToUpper(str[0]) + str.Substring(1);
+                return char.ToLower(str[0]) + str.Substring(1);
```

Comment:
I _think_ you want invariant culture conversion:

```suggestion
                return char.ToLowerInvariant(str[0]) + str.Substring(1);
```